src/app/(auth)/gifts/page.tsx

```diff
@@ -0,0 +1,99 @@
+import FiltersContainer from "~/app/_components/common/FiltersContainer";
+import { H4 } from "~/app/_components/common/H4";
+import { H6 } from "~/app/_components/common/H6";
+import ProductCard from "~/app/_components/common/ProductCard";
+import { api } from "~/trpc/server";
+import { type RouterOutputs } from "~/trpc/shared";
+
+type Review = RouterOutputs["gift"]["getFiltered"][number]["reviews"][number];
+
+export default async function Page({
+  searchParams,
+}: {
+  searchParams?: {
+    attributes?: string;
+    features?: string;
+    query?: string;
+    price?: string;
+  };
+}) {
+  const {
+    attributes: attributesFilter,
+    features: featuresFilter,
+    query: textFilter,
+    price,
+  } = searchParams ?? {};
+
+  const { attributes: attributesForCategory, features: featuresForCategory } =
+    await api.filter.getByCategoryId.query({
+      categoryId: 1,
+    });
+
+  const calculateRating = (reviews: Review[]) => {
+    if (reviews.length === 0) return 0;
+    const total: number = reviews.reduce(
+      (acc: number, review: Review) => acc + review.rating,
```

Comment:
It's not necessary to put the types here because TypeScript can infer them
```suggestion
      (acc, review) => acc + review.rating,
```